### DISCUSSION TITLE
can i get uinput.ko for miyoo mini?

### DISCUSSION THREAD

**Discussion by trngaje** [linux-chenxing/linux-chenxing.org#59] [General]
i want to use user space input driver.
but there is no uinput.ko in miyoo mini kernel module.
can i get uinput.ko ? or how to build uinput.ko.

**Comment by fifteenhex** [comment]
You would need to build a new kernel with the SDK sources currently.
Keep in mind the SDK kernel is 4.x so maybe it doesn't have the module you want.

Mainline does run on the miyoomini so you could build whatever modules you want but there is still a lot of work to make it nice to use.

**Comment by trngaje** [comment]
I downloaded the linux-4.9.84 source and built it to get uinput.ko. However, "insmod uinput.ko" causes an error called invalid format.
i cannot get working uinput.ko without defconfig file for miyoomini.

**Reply by fifteenhex** [reply]
I'm not sure what you want me to do.. I don't support the vendor kernels. You'll need to tell miyoo to give you the source if you need to build a module that loads into their kernel.

My long term plan is to get the miyoo mini fully working with mainline. The vendor kernels are a dead end.
